### PR TITLE
remove ticker.minds

### PR DESCRIPTION
**Pull request by adacovsk**
Noticed this when doing the roundstart stuff, I couldn't think of a reason why it would be necessary, since it's relatively easy to access the mind through a mob. While trivial, it does not need to exist, just adds to confusion, and by removing it, it's one less thing to worry about. It's used in three things: body archives (which I repurposed and works), scoreboard stats for survivors (it referenced mobs anyway, so rather than looping through minds, it loops through mobs), and targets (used `H.mind` to maintain its intent, where H is a human mob, which lessens the amount of checks).

It's just a not-well maintained list, and targets for objectives aren't even held in this list. This list just contains a list of minds, but why not just use mob_list? You can just access mind just as readily.

## What this does
- removes the existence of ticker.minds

## Why it's good
- Less junk that isn't futureproofed

### CODE REVIEW

**Comment by Eneocho** [comment]
Can't wait for something to break in unknowable ways.

**Comment by Theophrastus1** [comment]
PREPARE FOR UNFORESEEN CONSEQUENCES

**Comment by pedr69** [comment]
okay but does this fix dead AI's not showing up on the roundend screen?

**Comment by adacovsk** [comment]
> okay but does this fix dead AI's not showing up on the roundend screen?

bro... this has been around forever hasn't it? completely unrelated

**Comment by wotisjanitor** [comment]
THUMBS DOWN!!!!!!

**Comment by west3436** [comment]
can you show an actual performance increase from this? if not the risk to reward seems tilted

**Comment by Exxion** [comment]
>can you show an actual performance increase from this?

Assuredly not. However, it is worth cleaning up the code, *if* this PR doesn't break anything. I haven't looked at it closely yet and cannot claim to know whether or not this is the case.

**Comment by WindowsErrors** [comment]
bro lets just see what happens

**Comment by ConfusedCactus** [comment]
I am not in favor of Adam deleting anything that he doesn't know what it's for.

**Comment by WindowsErrors** [comment]
we may never know what it's for unless we delete it to see what happens

**Comment by adacovsk** [comment]
> I am not in favor of Adam deleting anything that he doesn't know what it's for.

The only thing I don't know understand is: Kanef code:
```		
for(var/datum/mind/M2 in ticker.minds) //see above, but for ghosts
			if(M2.current == M)
				return TRUE
```
The logic here doesn't make any sense and I fixed it by just doing M.mind, referencing the mob, rather than the mind.
And: `manifest_entries.Add(new /datum/stat/manifest_entry(M))` I didn't change this, it should work as intended, but all signs say it does nothing anyway.

**Comment by pedr69** [comment]
>kanef code

vs
>adacovsk unforeseen circumstance code

which is the lesser evil?
i'll do my part by both upvoting and downvoting

**Comment by Eneocho** [comment]
Unforeseen consequences adamcode is an experience. You get something unique on each gunked round caused by the unforeseen consequences. It might be the greater evil, but merge it anyway, at least the bugs will be funny to watch. If shit breaks we can always pull one from the collabs' book and revert if something went wrong.

File: code/datums/statistics/stat_helpers.dm [hunk 1]
```diff
@@ -155,15 +155,10 @@
...
-	for(var/datum/mind/M in ticker.minds)
-		// i'd like manifest stats, eventually, to be something that doesn't depend on
-		// all data being 'pristine' at round end; for instance, an ayy'd player
-		// will probably have a different entry than what they were at roundstart
-		// and that's gross
-		manifest_entries.Add(new /datum/stat/manifest_entry(M))
-		if(istype(M.current, /mob/living) && !M.current.isDead())
-			add_survivor_stat(M.current)
-
+	for(var/mob/living/M in player_list)
```
Comment (nervere) [inline code/datums/statistics/stat_helpers.dm:158]: This isn't the same thing as `for(var/datum/mind/M in ticker.minds)`. With this change, anyone who has logged out (thus being removed from player_list, which only tracks mobs with clients) will not show up on the scoreboard.

I recommend you change this to `mob_list` and add a check for a mind. At the same time, this kind of problem demonstrates why a global list of minds is a good thing, even if `ticker.minds` is poorly implemented/utilized.

File: code/datums/statistics/stat_helpers.dm [hunk 2]
```diff
@@ -155,15 +155,10 @@
...
+		manifest_entries.Add(new /datum/stat/manifest_entry(M.mind))
+		if(!M.isDead())
+			add_survivor_stat(M)
```
Comment (nervere) [inline code/datums/statistics/stat_helpers.dm:161]: This isn't the same thing as `for(var/datum/mind/M in ticker.minds)`. With this change, anyone who has logged out (thus being removed from player_list, which only tracks mobs with clients) will not show up on the scoreboard.

I recommend you change this to `mob_list` and add a check for a mind. At the same time, this kind of problem demonstrates why a global list of minds is a good thing, even if `ticker.minds` is poorly implemented/utilized.

**Comment by adacovsk** [comment]
> okay but does this fix dead AI's not showing up on the roundend screen?

On second thought, it might?

**Comment by Kammerjunk** [comment]
test this at least once and have someone whose body was destroyed show up on the scoreboard

**Comment by PrimeDSS13** [comment]
alternative to this PR: don't

**Comment by adacovsk** [comment]
> test this at least once and have someone whose body was destroyed show up on the scoreboard
image

Does this work for you? Deletion of the antag mob also works. Malf will be tested too.

**Comment by adacovsk** [comment]
> > okay but does this fix dead AI's not showing up on the roundend screen?
> 
> On second thought, it might?

yeah this fixes it

**Comment by Kammerjunk** [comment]
@pedr69 if that works, it has nothing to do with this PR

**Comment by ShiftyRail** [comment]
Stop doing thing at random. If you want to improve performance, look at a profiler. Roundstart can lag horribly now. I am not sure if your previous PRs worked.

**Comment by adacovsk** [comment]
> Stop doing thing at random. If you want to improve performance, look at a profiler. Roundstart can lag horribly now. I am not sure if your previous PRs worked.

this wasn't about performance, retard, it's about something that doesn't work and isn't used

**Comment by adacovsk** [comment]
> Stop doing thing at random. If you want to improve performance, look at a profiler. Roundstart can lag horribly now. I am not sure if your previous PRs worked.

Also, roundstart works great now. No lights off or naked for 10+ seconds. Not only that, I no longer crash.

**Comment by nervere** [comment]
It definitely works. The problem is that it has a use case your PR:
- failed to account for
- failed to make a better replacement for

File: code/controllers/subsystem/supply_shuttle.dm [hunk 3]
```diff
@@ -486,11 +486,8 @@ var/datum/subsystem/supply_shuttle/SSsupply_shuttle
 		M.angry = 0
 		M.apply_disguise()
 	for(var/mob/living/M in contents)
-		if(M.key || M.ckey) //only mobs that were never player controlled
+		if(M.key || M.ckey || M.mind) //only mobs that were never player controlled
```
Comment (ShiftyRail) [inline code/controllers/subsystem/supply_shuttle.dm:489]: This is not a neutral change

File: code/datums/gamemode/objectives/target/target.dm [hunk 4]
```diff
@@ -46,15 +46,15 @@
 
 /datum/objective/target/proc/get_targets()
 	var/list/targets = list()
-	for(var/datum/mind/possible_target in ticker.minds)
-		if(possible_target != owner && ishuman(possible_target.current) && (possible_target.current.z != map.zCentcomm) && (possible_target.current.stat != DEAD) && !(possible_target.assigned_role in bad_assassinate_targets))
-			targets += possible_target
+	for(var/mob/living/carbon/human/H in mob_list)
+		if(H.mind && (H.mind.key != owner.key) && (H.z != map.zCentcomm) && !H.isDead() && !(H.mind.assigned_role in bad_assassinate_targets))
```
Comment (ShiftyRail) [inline code/datums/gamemode/objectives/target/target.dm:50]: Objectively slower than the mind list look-up.

Comment (adacovsk) [reply code/datums/gamemode/objectives/target/target.dm:50]: Speed isn't the point

File: code/controllers/subsystem/supply_shuttle.dm (hunk 3 above)
Comment (adacovsk) [reply code/controllers/subsystem/supply_shuttle.dm:489]: Can you please elaborate

**Pull request closed without merging**